src/kube-runtime/test/test_image_checker.py

```diff
@@ -0,0 +1,109 @@
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+import sys
+import unittest
+
+import yaml
+
+sys.path.append("{}/../src/init.d".format(
+    os.path.split(os.path.realpath(__file__))[0]))
+import image_checker
+
+package_directory_com = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestImageChecker(unittest.TestCase):
+    def setUp(self):
+        try:
+            os.chdir(package_directory_com)
+        except:
+            pass
+
+    def test_image_without_username_tag(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_user_tag.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_without_tag(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_tag.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_not_exist(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_exist.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, False)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_local(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_local_registry.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_auth(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_auth.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
```

Comment:
same, you can use metaclass, here's [an example](https://stackoverflow.com/a/20870875)

Reply:
use decorator instead.